astropy/units/core.py

```diff
@@ -868,13 +868,22 @@ def get_converter(self, other, equivalencies=[]):
         UnitsError
             If units are inconsistent
         """
-        other = Unit(other)
 
-        try:
-            scale = self._to(other)
-        except UnitsError:
-            return self._apply_equivalences(
-                self, other, self._normalize_equivalencies(equivalencies))
+        if self is other:
+            # Shortcut
+            # TODO: Should this even bother with the conversion or just pass
```

Comment:
@embray - I don't particularly like this change -- it adds complexity when at least for larger arrays the main culprit in making things slow is that the conversion has to involve making a copy (the scale determination is very fast, especially if units are the same). Also, my feeling is that this is the wrong place to optimize this -- the converter shouldn't even be called when the units are the same.